src/glossary.rst

```diff
@@ -1555,16 +1601,16 @@ Glossary
       run, but ``bar:fail`` was not marked as an :term:`optional output`.
 
       User intervention is required to fix a stall, e.g. by retriggering
-      incomplete tasks after fixing the problems that caused them to fail.
+      tasks after fixing the problems that caused them to fail.
 
 
    suicide trigger
-      Suicide triggers remove :term:`tasks <task>` from the :term:`scheduler's
-      <scheduler>` active (:term:`n=0 <n-window>`) window at runtime.
+      Suicide triggers remove :term:`tasks <task>` from the 
+      :term:`n=0 window <n-window>` at runtime.
```

Comment:
Sorry, not sure I follow you?

Reply:
Since https://github.com/cylc/cylc-flow/pull/6067, suicide triggers don't *just* remove tasks from the task pool (aka the n=0 window), because the "removal" is indirectly recorded in the database (i.e. the task is considered "complete", but with a non-final status :thinking:).

I think a task that is suicide-triggered should be auto-spawned anyway so the n=0 bit is redundant. The limitation of suicide triggers is that they cant remove a task *after* it has become active.

Reply:
Oh, right, gotcha.  Well, perhaps it's kind of a matter of perspective. Only tasks in n=0 can cause a stall, so a task does not need to be suicided at all unless (a) it is gets spawned into n=0, and (b) would cause a stall if not removed.  So, they remove tasks from n=0, right?

> I think a task that is suicide-triggered should be auto-spawned anyway so the n=0 bit is redundant.

Do you mean auto-spawned by the suicide dependency? (and then, immediately removed by the suicide trigger).  If so, that's just an implementation trick to allow continued support for suicide triggers in places where they aren't actually needed at all in Cylc 8.